torch/_numpy/testing/utils.py

```diff
@@ -0,0 +1,2382 @@
+"""
+Utility function to facilitate testing.
```

Comment:
are these vendored from numpy or fresh?

Reply:
Vendored. Plus several fixes on top to account for numpy scalars vs 0D tensors etc.